src/Compilers/Core/Portable/Emit/EditAndContinue/DefinitionMap.cs

```diff
@@ -249,6 +250,31 @@ internal VariableSlotAllocator TryCreateVariableSlotAllocator(EmitBaseline basel
                 }
                 else
                 {
+                    // If the current method is async/iterator then either the previous method wasn't declared as async/iterator and it's updated to be one,
+                    // or it was but is not marked by the corresponding state machine attribute because it was missing in the compilation. 
+                    // In the later case we need to report an error since we don't known how to map to the previous state machine.
+
+                    // The IDE already checked that the attribute type is present in the base compilation, but didn't validate that it is well-formed.
+                    // We don't have the base compilation to directly query for the attribute, only the source compilation. 
+                    // But since constructor signatures can't be updated during EnC we can just check the current compilation.
+
+                    if (method.IsAsync)
+                    {
+                        if (compilation.CommonGetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_AsyncStateMachineAttribute__ctor) == null)
```

Comment:
BTW, we cast symbols to object before comparing with null

---
In reply to: [64441575](https://github.com/dotnet/roslyn/pull/11476#discussion_r64441575) [](ancestors = 64441575)